core/src/main/scala/quasar/destination/gbq/GBQDestinationModule.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.gbq
+
+import argonaut._, Argonaut._
+
+import cats.effect.{
+  Concurrent, 
+  ConcurrentEffect, 
+  ContextShift, 
+  Resource, 
+  Timer
+}
+import cats.data.EitherT
+import cats.implicits._
+
+import org.http4s.{
+  AuthScheme,
+  Credentials,
+  Method,
+  Request,
+  Status,
+  Uri}
+import org.http4s.headers.Authorization
+import org.http4s.client.Client
+import org.slf4s.Logging
+
+import quasar.api.destination.{DestinationType, DestinationError}
+import quasar.api.destination.DestinationError.InitializationError
+import quasar.connector.destination.{Destination, DestinationModule}
+import quasar.connector.MonadResourceErr
+
+import scala.Predef._
+import scala.{
+  StringContext,
+  Some,
+  Either,
+  Unit
+}
+import scala.concurrent.ExecutionContext
+
+object GBQDestinationModule extends DestinationModule with Logging {
+
+  val Redacted: String = "<REDACTED>"
+
+  def destinationType = DestinationType("gbq", 1L)
+
+  def sanitizeDestinationConfig(config: Json) = {
+    config.as[GBQConfig].toOption match {
+      //TODO: maybe redact the entire service account key
+      case Some(c) => c.copy(authCfg = c.authCfg.copy(privateKey = Redacted)).asJson
+      case _ => Json.jEmptyObject
+    }
+  }
+
+  def destination[F[_]: ConcurrentEffect: ContextShift: MonadResourceErr: Timer](config: Json)
+    : Resource[F,Either[InitializationError[Json], Destination[F]]] = {
+      val sanitizedConfig: Json = sanitizeDestinationConfig(config)
+      val configOrError = config.as[GBQConfig].toEither.leftMap {
+        case (err, _) =>
+          DestinationError.malformedConfiguration((destinationType, jString(Redacted), err))
+      }
+      val init = for {
+        cfg <- EitherT(Resource.pure[F, Either[InitializationError[Json], GBQConfig]](configOrError))
+        client <- EitherT(AsyncHttpClientBuilder[F](ConcurrentEffect[F], ExecutionContext.fromExecutor(null)).map(_.asRight[InitializationError[Json]]))
```

Comment:
`null`? 

How does this work?

Reply:
I'm not sure, but null executor is probably not the right thing to do. I'll make an executor and pass it it in.

Reply:
from [the docs](https://www.scala-lang.org/api/current/scala/concurrent/ExecutionContext$.html):
> the Executor to use. If null, a new Executor is created with default configuration.

Reply:
We'll want either a `newCachedThreadPool` or `newFixedThreadPool`, or both. Will loop back to this later.

Reply:
we didn't actually need a thread pool at all here, so deleted this